build-logic/src/main/kotlin/ktlint-kotlin-common.gradle.kts

```diff
@@ -39,7 +52,7 @@ tasks.withType<Test>().configureEach {
             (Runtime.getRuntime().availableProcessors() / 2).takeIf { it > 0 } ?: 1
         }
 
-    if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_16)) {
```

Comment:
1.`JavaVersion.current()` will always be 20, so I updated the code to read test java version from `javaLauncher` 

2. I decreased the condition to compare with Java 11, to avoid non-actionable warnings (which in Java 16 became errors and are the sole reason why additional `jvmArgs` are needed)